backbone/src/main/java/org/researchstack/backbone/ui/task/ReviewStepFullScreenImageActivity.kt

```diff
@@ -0,0 +1,65 @@
+package org.researchstack.backbone.ui.task
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import androidx.lifecycle.Observer
+import kotlinx.android.synthetic.main.rsb_activity_full_screen_image.*
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.core.parameter.parametersOf
+import org.researchstack.backbone.R
+import org.researchstack.backbone.step.Step
+import org.researchstack.backbone.ui.PinCodeActivity
+import org.researchstack.backbone.ui.step.fragments.ReviewStepFragment
+
+class ReviewStepFullScreenImageActivity : PinCodeActivity(), View.OnClickListener {
+    private val viewModel: ReviewStepFullScreenImageViewModel by viewModel { parametersOf(intent) }
+
+    companion object {
+        private const val EXTRA_STEP = "full_screen_step"
+        internal const val EXTRA_IMAGE_URL = "full_screen_image_url"
+        fun getCallingIntent(context: Context, step: Step, imageUrl: String): Intent {
+            val intent = Intent(context, ReviewStepFullScreenImageActivity::class.java)
+            intent.putExtra(EXTRA_STEP, step)
+            intent.putExtra(EXTRA_IMAGE_URL, imageUrl)
+            return intent
+        }
+
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.rsb_activity_full_screen_image)
+
+        initCallbacks()
+    }
+
+    private fun initCallbacks() {
+        viewModel.displayImageEvent.observe(this, Observer {
+            fullScreenImage.setImageBitmap(it)
+        })
+        fullScreenImageClose.setOnClickListener(this)
+        fullScreenImageEditStep.setOnClickListener(this)
+    }
+
+    override fun onClick(v: View?) {
```

Comment:
if you don't mind and because you're going to be making changes, can you rename this "v" to `view`.

Reply:
Personnally i prefer to avoid to do the onClick like that. It's to avoid to always to the when stuff.

I prefer for example to do something like that : 

fullScreenImageClose.setOnClickListener {
    setResult(Activity.RESULT_CANCELED)
    finish()
}
fullScreenImageEditStep.setOnClickListener {
    val step: Step? = intent.extras?.getSerializable(EXTRA_STEP) as Step?
    val data = Intent()
    data.putExtra(ReviewStepFragment.EXTRA_FULL_SCREEN_STEP, step)
    setResult(ReviewStepFragment.RESULT_FULL_SCREEN_EDIT, data)
    finish()
}

Reply:
I think both methods are fine, when you have to set up a bunch of click listeners, using "this" is preferred because in your example, you're creating two anonymous instances of the ClickListener interface, whereas if you set a common one, you only need one reference to the existing class _polymorphed_ as a ClickListener. 

Additionally, when they are related like in this case, it's a lot easier to find the code next to each other; I agree that it really depends. In this case, I like how he did it rather than individual ones, but I have no argument against either besides the irrelevant memory footprint and legibility in some cases. It usually also depends what you do inside the click listener. 

Personally, if I had to set up a bunch of click listeners and I wanted them inline/individually, I'd rather use method reference:

`view.setOnClickListener(this::doSomething)`
`view2.setOnClickListener(this::doSomethingElse)`

And have 
```
    private fun doSomething(view: View) {

    }

    private fun doSomethingElse(view: View) {

    }
```

Or alternatively: 

```
view.setOnClickListener(this::handleSomething)
view2.setOnClickListener(this::handleSomething)

    private fun handleSomething(view: View) {
      when (view..) 
            R.id.thisView -> doThis()
            R.id.thatView -> doThat()
    }
```

In short: it really depends :) What are your thoughts?

Reply:
I usually use `setOnClickListener(this)` when there are several views to click. It's exactly what Martin said about having one instance instead of having several.

Reply:
And can you rename it to `view` instead of `v`? :) Thanks!

(that was the initial comment)